src/worker.cpp

```diff
@@ -1101,13 +1103,101 @@ void Worker::populateDbus(const types::VPDMapVariant& parsedVpdMap,
+{
+    std::string l_assetTag;
+
+    try
+    {
+        // system VPD will be in IPZ format.
+        if (auto l_parsedVpdMap =
+                std::get_if<types::IPZVpdMap>(&i_parsedVpdMap))
+        {
+            auto l_itrToVsys = (*l_parsedVpdMap).find(constants::recVSYS);
+            if (l_itrToVsys != (*l_parsedVpdMap).end())
+            {
+                std::string l_tmKwdValue;
+                vpdSpecificUtility::getKwVal(l_itrToVsys->second,
+                                             constants::kwdTM, l_tmKwdValue);
+
+                std::string l_seKwdValue;
+                vpdSpecificUtility::getKwVal(l_itrToVsys->second,
+                                             constants::kwdSE, l_seKwdValue);
+
+                l_assetTag = std::string{"Server-"} + l_tmKwdValue +
+                             std::string{"-"} + l_seKwdValue;
+            }
+            else
+            {
+                throw std::runtime_error(
+                    "VSYS record not found in parsed VPD map to create Asset tag.");
+            }
+        }
+        else
+        {
+            throw std::runtime_error(
+                "Invalid VPD type recieved to create Asset tag.");
+        }
+    }
+    catch (const std::exception& l_ex)
+    {
+        // TODO:Log PEL with below description.
+        logging::logMessage("Asset tag can't be formed. Error = " +
+                            std::string(l_ex.what()));
+    }
+
+    return l_assetTag;
+}
+
 void Worker::publishSystemVPD(const types::VPDMapVariant& parsedVpdMap)
 {
     types::ObjectMap objectInterfaceMap;
 
     if (std::get_if<types::IPZVpdMap>(&parsedVpdMap))
     {
         populateDbus(parsedVpdMap, objectInterfaceMap, SYSTEM_VPD_FILE_PATH);
+
+        try
+        {
+            if (m_isFactoryResetDone)
+            {
+                const auto& l_assetTag = createAssetTagString(parsedVpdMap);
+
+                if (!l_assetTag.empty())
+                {
+                    auto l_itrToSystemPath =
+                        objectInterfaceMap.find(sdbusplus::message::object_path(
+                            constants::systemInvPath));
+                    if (l_itrToSystemPath == objectInterfaceMap.end())
+                    {
+                        throw std::runtime_error(
+                            "System Path not found in object map.");
+                    }
+                    else
+                    {
+                        types::PropertyMap l_assetTagProperty;
+                        l_assetTagProperty.emplace("AssetTag", l_assetTag);
+
+                        (l_itrToSystemPath->second)
+                            .emplace(constants::assetTagInf,
+                                     std::move(l_assetTagProperty));
+                    }
+                }
+                else
+                {
+                    throw std::runtime_error("Asset tag string is empty.");
```

Comment:
don't we need to still create the Asset tag object  and property even if we cannot fill the value ?

Reply:
Yes we should add, so that it can be updated via GUI.
I have updated the code to handle the empty scenario as well.